precompile/erc20/approve.go

```diff
@@ -135,7 +135,7 @@ func handleAuthorization(
 
 	if authorization == nil {
 		if amount.Sign() == 0 {
-			err = fmt.Errorf("no existing approvals, cannot approve 0")
+			err = nil // this is not an error to comply with the ERC20 std behavior.
```

Comment:
Note that `amoun.Sign() == 0` also covers the case where someone calls it with a negative value. We default to zero in that case in `ApproveMethod.Run`. Would be good to explicitly fail for negatives.

Reply:
Same for `PermitMethod` that uses `handleAuthorization`.

Reply:
I do not believe so as per the code?
```
   40// Sign returns:
   41//   - -1 if x < 0;
   42//   - 0 if x == 0;
   43//   - +1 if x > 0.
   44func (x *Int) Sign() int {
```

Reply:
addressed in: 03396387d80d58ae442524ccc8dced82489afae2